packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -596,7 +607,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     this.autoScalingGroup = new CfnAutoScalingGroup(this, 'ASG', asgProps);
     this.osType = imageConfig.osType;
-    this.autoScalingGroupName = this.autoScalingGroup.ref;
+    this.autoScalingGroupName = Token.isUnresolved(this.physicalName) ? this.autoScalingGroup.ref : this.physicalName;
```

Comment:
#We usually always return the `ref` in this case because when it’s used in the template it also encodes the implicit dependency. If we just return the concrete string, there won’t be an implicit dependency

Reply:
but what about the cross region referencing case?

Reply:
You need to use `this.getResourceName` and `getResourceArn` to obtain a polymorphic token for the name.

See other examples like s3 buckets.

Reply:
Thanks!